### PR TITLE
bug/empty-elements

### PR DESCRIPTION
**Pull request by mallorih**
- This PR adds a function to check if a piece of text only contains a bullet (no text) to prevent creating an empty element. 
- Also fixed a test that had a typo.

**Testing**
```Python
from unstructured.partition.text import partition_text
elements = partition_text('example-docs/fake-email.txt')

for ele in elements:
    print(ele)

```

### CODE REVIEW

File: unstructured/partition/text.py
Comment (badGarnet) [inline unstructured/partition/text.py:250]: can we then just pass in `clean_bullets(ctext)` to `element_from_text` and skip clean_bullets in the next function? Here we are calling the same regex pattern multiple times.

Comment (mallorih) [reply unstructured/partition/text.py]: So, I've moved the logic to `elements_from_text` but we can't remove the `clean_bullets` logic or maybe I'm misunderstood your suggestion.

File: unstructured/partition/text.py
Comment (cragwolfe) [inline unstructured/partition/text.py:267]:
```suggestion
                text=clean_text,
```
don't need to `clean_text` again?

Comment (mallorih) [reply unstructured/partition/text.py:267]: Yep that's what I meant! Thanks

**Review by cragwolfe — COMMENTED** [review]
Just needs a test? otherwise LGTM

File: unstructured/partition/text.py
Comment (qued) [inline unstructured/partition/text.py:270]: If the `if` statement fails, there's no fallback, so this function returns `None`.

**Review by qued — CHANGES_REQUESTED** [review]
I added a test that exposed an issue with the new logic.

File: CHANGELOG.md
Comment (qued) [inline CHANGELOG.md:50]: Remove this please.

File: CHANGELOG.md
Comment (qued) [inline CHANGELOG.md:375]: Remove this please.

File: unstructured/partition/text.py
Comment (qued) [inline unstructured/partition/text.py:160]: Could we call this something slightly more descriptive like maybe `is_not_empty_bullet`? Also needs a short docstring to clarify the purpose, and a type hint for the output.

**Review by qued — COMMENTED** [review]
A few minor changes needed, but the logic works for me. Ping me to approve once the changes are made.

File: unstructured/partition/text.py
Comment (qued) [reply unstructured/partition/text.py:160]: My reasoning for the name change is something like this is counterintuitive:
image

**Review by qued — APPROVED** [review]
LGTM!

**Pull request merged**
